google/cloud/forseti/services/model/importer/importer.py

```diff
@@ -239,8 +276,54 @@ def run(self):
             self.model.set_done(item_counter)
         finally:
             self.session.commit()
+            self.session.autocommit = autocommit
             self.session.autoflush = autoflush
 
+    @staticmethod
+    def model_action_wrapper(session,
+                             inventory_iterable,
+                             pre_action,
+                             action,
+                             post_action,
+                             flush_count,
+                             is_resource=False):
+        """Model action wrapper.
+
+        Args:
+            session (Session): Database session.
+            inventory_iterable (Iterable): Inventory iterable.
+            pre_action (func): Action taken before iterating the
+                inventory list.
+            action (func): Action taken during the iteration of
+                the inventory list.
+            post_action (func): Action taken after iterating the
+                inventory list.
+            flush_count (int): Flush every flush_count times.
+            is_resource (bool): If we are dealing with resources.
+        Returns:
+            int: Number of item iterated.
+        """
+
+        if pre_action:
+            pre_action()
+        item_counter = 0
+        ret_val = None
+        for inventory_data in inventory_iterable:
+            item_counter = item_counter + 1
+            if isinstance(inventory_data, tuple):
+                ret_val = action(*inventory_data)
+            elif not is_resource:
+                ret_val = action(inventory_data)
+            else:
+                ret_val = action(inventory_data, ret_val)
+            if not item_counter % flush_count:
+                session.flush()
+        if is_resource:
+            action(None, ret_val)
```

Comment:
qq: why is action() being called again for?  Why not include this in the `else` block above, which is for processing when `is_resource` is True?